projects/core/components/primitive-year-picker/primitive-year-picker.component.ts

```diff
@@ -63,7 +64,12 @@ export class TuiPrimitiveYearPickerComponent {
     get isSingle(): boolean {
         const {value} = this;
 
-        return !!value && this.isRange(value) && value.from.yearSame(value.to);
+        return (
+            !!value &&
+            !Array.isArray(value) &&
+            this.isRange(value) &&
```

Comment:
Same here:
```ts
    @HostBinding('class._single')
    get isSingle(): boolean {
        return this.isRange(this.value) && this.value.from.yearSame(this.value.to);
    }
```
Also `itemIsInterval` method can be refactored:
```ts
    itemIsInterval(item: number): boolean {
        const {value, hoveredItem} = this;

        if (!this.isRange(value) || !hoveredItem || value.from.year === hoveredItem) {
            return false;
        }

        return !value.from.yearSame(value.to)
            ? value.from.year <= item && value.to.year > item
            : tuiInRange(
                  item,
                  Math.min(value.from.year, hoveredItem),
                  Math.max(value.from.year, hoveredItem),
              );
    }
```
Or maybe even further, I haven't read into the ternary at the end, but `return false` typically means we can simplify it into a one big `return condition` 😄

Reply:
Actually looks like I'm wrong about that refactor, order there matters so let's keep it as is

Reply:
fixed